docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+
+Example usage of the GRPC API would look like:
+
+1. Starting a job:
+
+```go
+var client jobv1.JobClient
+// ...
+mcores := int64(1000) // 1 core
+jobId, err := client.Start(cmd.Context(), &jobv1.JobSpec{
+  Command: &jobv1.CommandSpec{
+    Command: "kubectl",
+    Args:    []string{"logs", "--follow", "pod/example"},
+    Env:     []string{"KUBECONFIG=/path/to/kubeconfig"},
+  },
+  Limits: &jobv1.ResourceLimits{
+    Cpu: &mcores,
+  },
+})
+```
+
+2. Streaming output:
+
+```go
+stream, err := client.Output(cmd.Context(), &jobv1.OutputRequest{
+  Id:     jobId,
+  Follow: true,
+})
+if err != nil {
+  return err
+}
+for {
+  resp, err := stream.Recv()
+  if err != nil {
+    return err
+  }
+  fmt.Fprint(os.Stdout, resp.GetOutput())
+}
+```
+
+3. Stopping the job:
+
+```go
+_, err := client.Stop(ctx, jobId)
+```
+
+#### CLI
+
+The CLI is the primary means of interacting with the Job Server. It will have a familiar interface to other CLI tools like `docker` or `kubectl`.
+The following examples demonstrate how the CLI will be used:
+
+```bash
+# starting the job server
+$ sudo jobserver serve --listen-address=127.0.0.1:9090 --rbac=rbac-config.yaml \
+  --cacert=ca.crt --cert=tls.crt --key=tls.key
+
+# to use the client, flags to configure cert and key paths, and the server address, are required.
+# these are omitted in the remaining examples for brevity.
+$ jobserver --address=127.0.0.1:9090 --cacert=ca.crt --cert=tls.crt --key=tls.key ...
+
+# starting a job:
+# for commands that don't require flag args, they can be passed as-is:
+$ jobserver run kubectl logs pod/example
+<job-id> # job id is printed
+
+# for commands that do require flag args, the expected '--' delimiter can be used:
+$ jobserver run -- kubectl logs --follow pod/example
+<job-id>
+
+# to start a job with resource limits:
+$ jobserver run \
+  --cpus=100m \ # 100 milli-cores; can also write e.g. '--cpus=2' for 2 cores (no float values)
+  --memory=1Gi --memory-soft-limit=512Mi \ # the familiar suffixes can be used for memory limits
+  --device-read-bps=8:16=200 \ # format is major:minor=value (bytes per second)
+  --device-write-bps=8:16=200,8:0=50 \ # multiple devices can be specified by separating with commas
+  --device-read-iops=... --device-write-iops=... \ # same format, but units are in iops
+  -- go build -o bin/jobserver ./cmd/jobserver
+
+# stopping a job:
+$ jobserver stop <job-id>
+<job-id>
+
+# getting the status of a job:
+$ jobserver status <job-id>
+state: running
+pid: 1234
+start time: 2006-01-02T15:04:05Z07:00
+
+# getting the output of a job:
+$ jobserver logs [--follow] <job-id>
+<output>
+
+# getting a list of all job ids:
+$ jobserver list
+JOB ID      COMMAND       CREATED          STATUS
+<id-1>      kubectl       2006-01-02...    Running
+<id-2>      go            2006-01-02...    Completed
```

Comment:
nit: 

Right now the `list` API returns only  `JobIdList` (`rpc List(google.protobuf.Empty) returns (JobIdList)`) 
where additional information needs to be fetch sperattly from Status endpoint (`rpc Status(JobId) returns (JobStatus)`)

Does it make sense to return basic job information about a job from the `List` endpoint ?

Reply:
It could work, but it would mean the List and Status methods have overlapping functionality. Also, if there are a very large number of jobs, calling List() could potentially become a very expensive operation for the server, and would make rate limiting less effective if that were to be added.

Reply:
All right. lets leave the current version.